File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        Set<Integer> set = new HashSet<>();

        for(int num : nums1) {
            set.add(num);
        }

        List<Integer> intersection = new ArrayList<>();

        for(int num : nums2) {
            if(set.contains(num)) {
                intersection.add(num);
                set.remove(num);
            }
        }
        int[] ans = new int[intersection.size()];

        int i = 0;
        for(int num : intersection) {
            ans[i++] = num;
        }
        return ans;
    }
}

File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.py
class Solution(object):
    def intersection(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        sets = set()

        for num in nums1:
            sets.add(num)
        intersection = []

        for num in nums2:
            if(num in sets) :
                intersection.append(num)
                sets.remove(num)

        return intersection
        